public/js/app/validation.js

```diff
@@ -0,0 +1,4 @@
+$("input[name='username']").change(function(event){$.getJSON("https://api.faforever.com/data/player?filter[player]=login==%27"+ $("input[name='username']").val() +"%27",
+   function(data) {
+     if(!jQuery.isEmptyObject(data.data)){alert("username is already in use");}
+   });})
```

Comment:
Please, format it properly + some comments:
```javascript
$("input[name='username']").change(
	function(event) {
		$.getJSON(
			// https://api.faforever.com/data should be in constants as path
			// getJSON accepts data as second parameter, did you try it?
			"https://api.faforever.com/data/player?filter[player]=login==%27" + $("input[name='username']").val() + "%27",
			function(data) { // i would rename data to result, because it's xhr result object.
				if(!jQuery.isEmptyObject(data.data)) {
					// I think, you can inform user much better, not with alert.
					alert("username is already in use");
				}
			}
		);
	}
);
```

Reply:
Also on alert you should focus user back to input.

Reply:
updated the file accordingly